paimon-common/src/main/java/org/apache/paimon/fs/ExternalPathProvider.java

```diff
@@ -0,0 +1,213 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.fs;
+
+import org.apache.paimon.CoreOptions.ExternalPathStrategy;
+import org.apache.paimon.annotation.VisibleForTesting;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/** Provider for external paths. */
+public class ExternalPathProvider implements Serializable {
+    private static final String S3 = "s3";
+    private static final String OSS = "oss";
+
+    private final Map<String, Path> externalPathsMap;
+    private final List<Path> externalPathsList;
+
+    private final ExternalPathStrategy externalPathStrategy;
+    private final String externalFSStrategy;
+    private int currentIndex;
+    private boolean externalPathExists;
+    private final String dbAndTableRelativePath;
+
+    @VisibleForTesting
+    public ExternalPathProvider() {
+        this.externalPathsMap = new HashMap<>();
+        this.externalPathsList = new ArrayList<>();
+        this.externalPathStrategy = ExternalPathStrategy.NONE;
+        this.externalFSStrategy = null;
+        this.dbAndTableRelativePath = null;
+    }
+
+    public ExternalPathProvider(
+            String externalPaths,
+            ExternalPathStrategy externalPathStrategy,
+            String externalFSStrategy,
+            String dbAndTableRelativePath) {
+        this.externalPathsMap = new HashMap<>();
+        this.externalPathsList = new ArrayList<>();
+        this.externalPathStrategy = externalPathStrategy;
+        this.externalFSStrategy = externalFSStrategy;
+        this.dbAndTableRelativePath = dbAndTableRelativePath;
+        this.currentIndex = 0;
+        initExternalPaths(externalPaths);
+    }
+
+    private void initExternalPaths(String externalPaths) {
+        if (externalPaths == null) {
+            return;
+        }
+
+        if (externalPathStrategy != null
+                && externalPathStrategy.equals(ExternalPathStrategy.SPECIFIC_FS)) {
+            if (externalFSStrategy == null) {
+                throw new IllegalArgumentException("external fs strategy should not be null: ");
+            }
+        }
+
+        String[] tmpArray = externalPaths.split(",");
+        for (String part : tmpArray) {
+            String path = part.trim();
+            if (path.toLowerCase().startsWith(OSS)) {
```

Comment:
You don't need to bind oss or s3, you can just pick the paths by starting with SpecificFS.